src/main/java/duke/storage/Storage.java

```diff
@@ -0,0 +1,86 @@
+package duke.storage;
+
+import duke.DukeException;
+import duke.tasks.*;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Stores the user data in a text file.
+ */
+public class Storage {
+    private final File directory = new File("data");
+    private final File file = new File("data/tasks.txt");
+
+    /**
+     * Constructs a storage pointing to a text file.
+     *
+     * @throws DukeException If there is error when creating file.
+     */
+    public Storage() throws DukeException {
+        if (!directory.exists()) {
+            directory.mkdir();
+        }
+
+        try {
+            file.createNewFile();
+        } catch (IOException e) {
+            throw new DukeException("Error creating storage file! You data will not be stored.", e);
+        }
+    }
+
+    /**
+     * Retrieves the tasks store in the storage text file.
+     *
+     * @return The tasks stored in an array list.
+     */
+    public ArrayList<Task> retrieveTaskList() {
```

Comment:
Perhaps to better match the name of the variable you return, your method name can be retrieveTasks